application/controllers/ConfigController.php

```diff
@@ -135,44 +141,78 @@ function (NotificationsConfigForm $form) use ($kconfig, $sourceForm) {
                                 }
 
                                 $db->insert((new KConfig())->getTableName(), [
-                                    $db->quoteIdentifier('key') => $k,
-                                    'value'                     => $v,
+                                    'cluster_uuid' => Uuid::fromString($clusterUuid)->getBytes(),
+                                    $key           => $k,
+                                    'value'        => $v
                                 ]);
                             }
                         });
                     } catch (Throwable $e) {
                         Logger::error($e);
                         Logger::error($e->getTraceAsString());
 
-                        $form->addMessage($e->getMessage());
-
-                        return;
+                        throw $e;
                     }
 
                     Notification::success(
                         $this->translate('New configuration has successfully been stored.')
                     );
 
-                    $this->redirectNow('__REFRESH__');
+                    $this->redirectNow(Url::fromPath('kubernetes/config/notifications', ['id' => $clusterUuid]));
                 }
             )->handleRequest($this->getServerRequest());
 
-        if (
-            preg_match(
-                '/source-(\d+)/',
-                $kconfig[KConfig::NOTIFICATIONS_USERNAME]->value ?? '',
-                $matches
-            ) !== false
-            && ! empty($matches)
-        ) {
-            try {
-                $sourceForm->loadSource($matches[1]);
-
-                // TODO(el): Check password mismatch.
-            } catch (HttpNotFoundException $e) {
-                // TODO(el): Add error box.
-            }
-        }
+        $this->addContent($form);
+    }
+
+    public function prometheusAction()
+    {
+        $form = (new PrometheusConfigForm())
+            ->populate(['cluster_uuid' => $this->getRequest()->get('id')])
```

Comment:
Please rename `id` to `cluster_uuid` and get the default value from the session.